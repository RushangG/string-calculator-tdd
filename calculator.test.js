const { add } = require('./calculator');

test('empty string returns 0', () => {
  expect(add('')).toBe(0);
});

test('single number returns its value', () => {
  expect(add('5')).toBe(5);
});

test('two numbers, comma delimited, returns the sum', () => {
  expect(add('1,2')).toBe(3);
});

test('supports custom delimiter', () => {
  expect(add('//;\n1;2')).toBe(3);
});


test('throws on negative numbers with message', () => {
  expect(() => add('1,-2,-3')).toThrow('negative numbers not allowed: -2,-3');
});