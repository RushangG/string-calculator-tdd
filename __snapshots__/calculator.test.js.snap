// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`empty string returns 0 1`] = `0`;

exports[`ignores numbers larger than 1000 1`] = `2`;

exports[`ignores numbers larger than 1000 2`] = `1001`;

exports[`multiple numbers, comma delimited 1`] = `10`;

exports[`numbers separated by newlines and commas 1`] = `6`;

exports[`single number returns its value 1`] = `5`;

exports[`supports custom delimiter 1`] = `3`;

exports[`supports custom multi-character delimiter 1`] = `6`;

exports[`supports custom single-character delimiter 1`] = `3`;

exports[`supports multiple custom delimiters 1`] = `6`;

exports[`supports multiple multi-character delimiters 1`] = `6`;

exports[`two numbers, comma delimited, returns the sum 1`] = `3`;
